Wrappers/Python/cil/optimisation/algorithms/Algorithm.py

```diff
@@ -276,8 +276,9 @@ def run(self, iterations=None, callbacks: Optional[List[Callback]]=None, verbose
         self.max_iteration = self.iteration + iterations
         iters = (count(self.iteration) if np.isposinf(self.max_iteration)
                  else range(self.iteration, self.max_iteration))
-        for _ in zip(iters, self):
+        for _ in iters:
             try:
+                self.__next__()
```

Comment:
The equivalent would be with `__next__` called outside the `try/except` statement.

Reply:
Looks like it isn't!